plugins/woocommerce/includes/class-wc-order.php

```diff
@@ -392,7 +392,23 @@ protected function status_transition() {
 
 		if ( $status_transition ) {
 			try {
-				do_action( 'woocommerce_order_status_' . $status_transition['to'], $this->get_id(), $this );
+				/**
+				 * Fires when order status is changed.
+				 *
+				 * @since x.x.x Added $status_transition parameter.
+				 *
+				 * @param int Order ID.
+				 * @param WC_Order $order Order object.
+				 * @param array $status_transition {
+				 * 		Status transition data.
```

Comment:
There are some [spacing issues](https://github.com/woocommerce/woocommerce/actions/runs/6657976273/job/18528394174?pr=41051) in this section of the code preventing our tests from completing successfully. See:

```
 403 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (WordPress.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
 405 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (WordPress.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
 406 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (WordPress.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
 407 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (WordPress.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
 408 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (WordPress.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
```